logic/src/commonMain/kotlin/com/wire/kalium/logic/data/asset/AssetRepository.kt

```diff
@@ -22,19 +24,28 @@ internal class AssetDataSource(
     private val assetDao: AssetDAO
 ) : AssetRepository {
 
-    override suspend fun uploadAndPersistPublicAsset(uploadAssetData: UploadAssetData): Either<NetworkFailure, UploadedAssetId> =
-        suspending {
-            wrapApiRequest {
-                // we should also consider for images, the compression for preview vs complete picture
-                assetMapper.toMetadataApiModel(uploadAssetData).let { metaData ->
-                    assetApi.uploadAsset(metaData, uploadAssetData.data)
-                }
-            }.map { assetResponse ->
-                val assetEntity = assetMapper.fromUploadedAssetToDaoModel(uploadAssetData, assetResponse)
-                assetDao.insertAsset(assetEntity)
-                assetMapper.fromApiUploadResponseToDomainModel(assetResponse)
+    override suspend fun uploadAndPersistPublicAsset(mimeType: AssetType, assetData: ByteArray): Either<NetworkFailure, UploadedAssetId> {
+        val uploadAssetData = UploadAssetData(assetData, mimeType, true, RetentionType.ETERNAL)
+        return doUploadAndPersistAsset(uploadAssetData)
+    }
+
+    override suspend fun uploadAndPersistPrivateAsset(mimeType: AssetType, assetData: ByteArray): Either<CoreFailure, UploadedAssetId> {
+        val uploadAssetData = UploadAssetData(assetData, mimeType, false, RetentionType.PERSISTENT)
+        return doUploadAndPersistAsset(uploadAssetData)
+    }
+
+    private suspend fun doUploadAndPersistAsset(uploadAssetData: UploadAssetData): Either<NetworkFailure, UploadedAssetId> = suspending {
```

Comment:
```suggestion
    private suspend fun uploadAndPersistAsset(uploadAssetData: UploadAssetData): Either<NetworkFailure, UploadedAssetId> = suspending {
```


I would drop the `do` prefix, `upload` is already a verb.